Source/iop/Iop_SubSystem.cpp

```diff
@@ -184,11 +184,10 @@ uint32 CSubSystem::ReadIoRegister(uint32 address)
 	{
 		return m_spu.ReadRegister(address);
 	}
-	else if(address >= CDmac::DMAC_ZONE1_START && address <= CDmac::DMAC_ZONE1_END)
-	{
-		return m_dmac.ReadRegister(address);
-	}
-	else if(address >= CDmac::DMAC_ZONE2_START && address <= CDmac::DMAC_ZONE2_END)
+	else if(
+	    address >= CDmac::DMAC_ZONE1_START && address <= CDmac::DMAC_ZONE1_END ||

```

Comment:
Please add parenthesis around expressions.

`address >= CDmac::DMAC_ZONE1_START && address <= CDmac::DMAC_ZONE1_END` 

should be

`(address >= CDmac::DMAC_ZONE1_START && address <= CDmac::DMAC_ZONE1_END)`

Reply:
Done